api/views/user_views.py

```diff
@@ -647,8 +647,38 @@ class DeleteUserView(APIView):
     def delete(self, request):
         id = request.user.id
         user = get_object_or_404(User, id=id)
-        user.delete()
-        return JsonResponse({"message": "User deleted successfully"})
+
+        if hasattr(user, "instructor"):
+            boto3_client = boto3.client(
+                "s3",
+                aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
+                aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY,
+                region_name=settings.AWS_S3_REGION_NAME,
+            )
+            bucket_name = settings.AWS_STORAGE_BUCKET_NAME
+            recordings = InstructorRecordings.objects.filter(instructor=user.instructor)
+
+            try:
+                for recording in recordings:
+                    boto3_client.delete_object(Bucket=bucket_name, Key=recording.s3_path)
+            except Exception as e:
+                return JsonResponse(
+                    {"error": f"Failed to delete files from S3: {str(e)}"},
+                    status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                )
```

Comment:
How about we delete the entire folder/object starting with `user_`? All recordings for users are stored in the folder/object `user_<user_id>`

Also, if that fails we should log that and we shouldn't return/exit; we don't have logging in place yet, so maybe we can add a `TODO` line to add a logging statement right after we implement the logging logic.

Reply:
I suppose we can; early exit will prevent users from deleting their accounts, which might be frustrating; the logging will flag to someone to do the deletion manually right after.

This can be mentioned in the privacy policy for full transparency. If you strongly feel that we should exit then that's okay too; however, we need to make sure that the `delete_object` doesn't throw an error if files/folder/objects do not exist.